### DISCUSSION TITLE
How to import a component with `dynamic` immediately instead of conditionally

### DISCUSSION THREAD

**Discussion by heloineto** [vercel/next.js#59978] [Help] [answered]
### Summary

I'm optimizing the First Load JS of my Next.js 14 app, the code is something like:

```jsx
const ExpansiveTable = dynamic(
  () => import('../expensive-table'),
);

export default function Page() {
  // fetches items in the API to display in the table
  const itemsQuery = useQuery(...);
  
  if(itemsQuery.isPending) {
    return <>loading...</>
  }
  
  if(itemsQuery.isError) {
    return <>error</>
  }
  
  return <ExpansiveTable items={itemsQuery.data} />
}
```

The problem with this code is that it creates a network waterfall, because it has to wait for the items request to finish to start fetching the component.

Since I know that in 99% of the cases the table will be eventually shown, it would be optimal to start fetching the component in parallel to the API request.

How do I do that?

**Comment by matthewraaff** [comment]
You could try disabling server side rendering in the settings of your ExpansiveTable.

```js
//example
const ExpansiveTable = dynamic(
  () => import('../expensive-table').then((module) => module.default),
  { ssr: false }
);
```

**Reply by heloineto** [reply]
I don't think that works, because it does not change when the code is fetched

**Reply by matthewraaff** [reply]
Hi, sorry for this issue. I've updated the code's syntax after reading the docs, I misinterpreted them originally.

I think disabling SSR should work by letting the client load it parallel, rather than linearly like it would if SSR was enabled. It causes a waterfall/queue-ish situation where the dynamic waits for it to be processed before anything is delivered.

If there are any nextjs experts please correct me if I'm wrong!

Have a great evening,
Matthew

**Comment by CalebBarnes** [comment, marked as answer]
```tsx
const ExpansiveTableWrapper = dynamic(
  () => import('../expensive-table-wrapper'),
);

export default function Page() {
  // fetches items in the API to display in the table
  const itemsQuery = useQuery(...);
  
  return <ExpansiveTableWrapper itemsQuery={itemsQuery} />
}
```

Inside ExpansiveTableWrapper you can add your loading and error states. 🚀

This way you are immediately rendering that dynamic component no matter if it is loading or not. Just import whatever you want inside that wrapper and those will be in the bundle of that dynamic wrapper component.

**Reply by heloineto** [reply]
That's a great idea for my use case. Thanks a lot!
